dace/codegen/tools/type_inference.py

```diff
@@ -355,17 +356,25 @@ def _BinOp(t, symbols, inferred_symbols):
 
 
 def _Compare(t, symbols, inferred_symbols):
-    _dispatch(t.left, symbols, inferred_symbols)
+    # If any vector occurs in the comparision, the inferred type is a bool vector
+    inf_type = _dispatch(t.left, symbols, inferred_symbols)
+    any_vector = isinstance(inf_type, dtypes.vector)
     for o, e in zip(t.ops, t.comparators):
         if o.__class__.__name__ not in cppunparse.CPPUnparser.cmpops:
             continue
-        _dispatch(e, symbols, inferred_symbols)
+        inf_type = _dispatch(e, symbols, inferred_symbols)
+        any_vector = any_vector or isinstance(inf_type, dtypes.vector)
 
+    return dtypes.vector(dace.bool, -1) if any_vector else dtypes.bool
```

Comment:
For the vector length, this is right.
In my opinion, the return type should still be a bool vector, because the result of a comparison is a boolean, right?

Reply:
Also regarding the length, the inference is used everywhere in DaCe and the SVE_LEN symbol might pop up unexpectedly in other codes.

Reply:
Makes sense with the vector length. In a vector comparision, the result type should be a bool vector because in the scalar case it a bool as well, right?